instrumentation-genai/opentelemetry-instrumentation-openai-agents/src/opentelemetry/instrumentation/openai_agents/__init__.py

```diff
@@ -0,0 +1,131 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+OpenAI Agents instrumentation supporting `openai` in agent frameworks.
+Enable with ``OpenAIAgentsInstrumentor``.
+
+.. _openai: https://pypi.org/project/openai/
+
+Usage
+-----
+
+.. code:: python
+
+    from openai import OpenAI
+    from opentelemetry.instrumentation.openai_agents import (
+        OpenAIAgentsInstrumentor,
+    )
+
+    OpenAIAgentsInstrumentor().instrument()
+
+    # Your OpenAI agents code here
+    client = OpenAI()
+
+API
+---
+
+This instrumentation captures spans for OpenAI API calls made within
+agent frameworks.
+It provides detailed tracing information including:
+
+- Request and response content (if configured)
+- Token usage metrics
+- Model information
+- Error handling
+
+Configuration
+-------------
+
+Environment variables:
+
+- ``OTEL_INSTRUMENTATION_OPENAI_AGENTS_CAPTURE_CONTENT``: ``true`` to
+    capture request/response content (default ``false``).
+- ``OTEL_INSTRUMENTATION_OPENAI_AGENTS_CAPTURE_METRICS``: ``true`` to
+    capture metrics (default ``true``).
+"""
+
+import logging
+from typing import Collection
+
+from agents.tracing import add_trace_processor
+
+from opentelemetry._events import get_event_logger
+from opentelemetry.instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.instrumentation.openai_agents.constants import (
+    GenAIEvaluationAttributes,
+    GenAIOperationName,
+    GenAIOutputType,
+    GenAIProvider,
+    GenAIToolType,
+)
+from opentelemetry.instrumentation.openai_agents.genai_semantic_processor import (
+    GenAISemanticProcessor,
+)
+from opentelemetry.instrumentation.openai_agents.package import _instruments
+from opentelemetry.semconv.schemas import Schemas
+from opentelemetry.trace import get_tracer
+
+logger = logging.getLogger(__name__)
+
+
+class OpenAIAgentsInstrumentor(BaseInstrumentor):
+    """An instrumentor for OpenAI Agents frameworks."""
+
+    def _instrument(self, **kwargs):
+        """Instruments the OpenAI library for agent frameworks."""
+        # TODO: Implement your instrumentation logic here
+        # This is a stub implementation - add your custom instrumentation code
+
+        tracer_provider = kwargs.get("tracer_provider")
+        tracer = get_tracer(
+            __name__,
+            "",
+            tracer_provider,
+            schema_url=Schemas.V1_28_0.value,
+        )
+        event_logger_provider = kwargs.get("event_logger_provider")
```

Comment:
@aabmass  @DylanRussell  @xrmx 

What is our stance on adding net new components that rely on the soon-to-be deprecated [events api/sdk[(https://github.com/open-telemetry/opentelemetry-python/pull/4654)?

Reply:
Ya I believe it should just use logs directly